src/main/java/com/fundynamic/d2tm/game/map/Map.java

```diff
@@ -98,21 +99,32 @@ public Shroud getShroud() {
     }
 
     public Cell getCellByAbsolutePixelCoordinates(int pixelX, int pixelY) {
-        return getCell(pixelX / TILE_SIZE, pixelY / TILE_SIZE);
+        return getCellProtected(pixelX / TILE_SIZE, pixelY / TILE_SIZE);
+    }
+
+    public Vector2D getCellCoordinatesInAbsolutePixels(int cellX, int cellY) {
+        return Vector2D.create(cellX * TILE_SIZE, cellY * TILE_SIZE);
     }
 
     public Unit placeUnit(Unit unit) {
         Vector2D mapCoordinates = unit.getMapCoordinates();
         getCell(mapCoordinates).setEntity(unit);
+        int range = Random.getRandomBetween(2, 5); // TODO: get this from unit!
+        revealShroudFor(mapCoordinates.getXAsInt(), mapCoordinates.getYAsInt(), range); // TODO: test this
         return unit;
     }
 
     public Structure placeStructure(Structure structure) {
         Vector2D topLeftMapCoordinates = structure.getMapCoordinates();
+        int range = 2; // TODO: get this from structure!
```

Comment:
Not sure if you're still working on this PR, but I think this TODO is part of it. `range` should probably be a property of `Entity`.